src/lib/support/BitFlags.h

```diff
@@ -102,7 +102,7 @@ class BitFlags
      */
     BitFlags & Clear(const BitFlags & other)
     {
-        mValue &= ~other.mValue;
+        mValue &= static_cast<IntegerType>(~static_cast<IntegerType>(other.mValue));
```

Comment:
Is there a reason why you've used `static_cast` twice?

Reply:
Right, I think only the outer cast is needed. This was not failing on my machine, only on CI and the only way to test for me was by pushing the code. So I went with "better copy a tested code" from the next Clear() on line 114.

Reply:
Even something like:

```cpp
void TestClear(nlTestSuite * inSuite, void * inContext)
{
    BitMask<TestEnum> mask1;
    BitMask<TestEnum> mask2;

    mask1.Set(TestEnum::kBit_0);
    mask1.Set(TestEnum::kBits_1_2);
    mask1.Set(TestEnum::kBits_High8);

    mask2.set(TestEnum::kBits_1_2);
    mask1.Clear(mask2);

    NL_TEST_ASSERT(inSuite, mask.Raw() == 0xFF01);
}
```

in TestBitMask seems to trigger the error and could be used as a test.

Reply:
@andy31415 have a look at https://github.com/project-chip/connectedhomeip/pull/30689

Reply:
That change should be part of this PR - we should commit code with tests together.

Reply:
Ah, ok. Will update it.